lib/chef/resource/support/client.erb

```diff
@@ -0,0 +1,64 @@
+<% %w(node_name
+      chef_license
+      chef_server_url
+      event_loggers
+      file_backup_path
+      file_cache_path
+      file_staging_uses_destdir
+      formatters
+      http_proxy
+      https_proxy
+      ftp_proxy
+      log_level
+      log_location
+      minimal_ohai
+      named_run_list
+      no_proxy
+      ohai_disabled_plugins
+      ohai_optional_plugins
+      pid_file
+      policy_group
+      policy_name
+      run_path
+      ssl_verify_mode).each do |prop| -%>
+<% next if eval('@' + prop).nil? || eval('@' + prop).empty? -%>
+<%=prop %> = <%= eval('@' + prop).inspect %>
+<% end -%>
+<%# log_location is special due to STDOUT/STDERR from String -> IO Object %>
+<% unless @log_location.nil? %>
+  <% if @log_location.is_a? String && %w(STDOUT STDERR).include?(@log_location) -%>
+log_location = <%= @log_location %>
+  <% else -%>
+log_location = <%= @log_location.inspect %>
+  <% end -%>
+<% end -%>
```

Comment:
Could you move this logic to a function in the resource and avoid the nested conditional here?

Reply:
I moved the actual creation of the handler string into the resource itself.